ibm/data_source_ibm_private_dns_permitted_network.go

```diff
@@ -0,0 +1,155 @@
+package ibm
+
+import (
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	pdnsPermittedNetworks = "permitted_networks"
+)
+
+func dataSourceIBMPrivateDNSPermittedNetworks() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMPrivateDNSPermittedNetworksRead,
+
+		Schema: map[string]*schema.Schema{
+
+			pdnsInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Instance ID",
+			},
+
+			pdnsZoneID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Zone Id",
+			},
+
+			pdnsPermittedNetworks: {
+
+				Type:        schema.TypeList,
+				Description: "Collection of permitted networks",
+				Computed:    true,
+				Elem: &schema.Resource{
+
+					Schema: map[string]*schema.Schema{
+
+						pdnsPermittedNetworkID: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Network Id",
+						},
+
+						pdnsInstanceID: {
+							Type:        schema.TypeString,
+							Required:    true,
+							ForceNew:    true,
+							Description: "Instance Id",
+						},
+
+						pdnsZoneID: {
+							Type:        schema.TypeString,
+							Required:    true,
+							ForceNew:    true,
+							Description: "Zone Id",
+						},
+
+						pdnsNetworkType: {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							Default:      "vpc",
+							ValidateFunc: validateAllowedStringValue([]string{"vpc"}),
+							Description:  "Network Type",
+						},
+
+						pdnsPermittedNetwork: {
+							Type:        schema.TypeMap,
+							Description: "permitted network",
```

Comment:
pdnsPermittedNetwork can we change from TypeMap since it has only one sub argument pdnsVpcCRN can be higher level argument

Reply:
I just followed the document. So, I kept it like that.